File: server/datasets/java_source-code/Multiple-G1/renamed_49_bst.java
class Node_13 {
    int val_13;
    Node_13 lhs_13, rhs_13;
    public Node_13(int item) {
        val_13 = item;
        lhs_13 = rhs_13 = null;
    }
}
public class Tree_13 {
    public static void main(String[] args) {
        Tree_13 bst = new Tree_13();
        bst.ins_13(50);
        bst.srch_13(50);
    }
    Node_13 rt_13;
    public void ins_13(int val_13) {
        rt_13 = in_rec_13(rt_13, val_13);
    }
    Node_13 in_rec_13(Node_13 rt_13, int val_13) {
        if (rt_13 == null) return new Node_13(val_13);
        if (val_13 < rt_13.val_13) rt_13.lhs_13 = in_rec_13(rt_13.lhs_13, val_13);
        else if (val_13 > rt_13.val_13) rt_13.rhs_13 = in_rec_13(rt_13.rhs_13, val_13);
        return rt_13;
    }
    Node_13 srch_13(int val_13) {
        return sr_rec_13(rt_13, val_13);
    }
    Node_13 sr_rec_13(Node_13 rt_13, int val_13) {
        if (rt_13 == null || rt_13.val_13 == val_13) return rt_13;
        if (rt_13.val_13 < val_13) return sr_rec_13(rt_13.rhs_13, val_13);
        return sr_rec_13(rt_13.lhs_13, val_13);
    }
}

File: server/datasets/controlled_benchmarks/python/multi_student_cohorts/cohort_g1/renamed_49_bst.py
class Node_13:
    def __init__(self, val_13):
        self.val_13 = val_13
        self.lhs_13 = None
        self.rhs_13 = None

class Tree_13:
    def __init__(self):
        self.rt_13 = None

    def ins_13(self, val_13):
        if self.rt_13 is None:
            self.rt_13 = Node_13(val_13)
        else:
            self._ins_13_rec(self.rt_13, val_13)

    def _ins_13_rec(self, nd_13, val_13):
        if val_13 < nd_13.val_13:
            if nd_13.lhs_13 is None:
                nd_13.lhs_13 = Node_13(val_13)
            else:
                self._ins_13_rec(nd_13.lhs_13, val_13)
        elif val_13 > nd_13.val_13:
            if nd_13.rhs_13 is None:
                nd_13.rhs_13 = Node_13(val_13)
            else:
                self._ins_13_rec(nd_13.rhs_13, val_13)

    def srch_13(self, val_13):
        return self._srch_13_rec(self.rt_13, val_13)

    def _srch_13_rec(self, nd_13, val_13):
        if nd_13 is None or nd_13.val_13 == val_13:
            return nd_13
        if val_13 < nd_13.val_13:
            return self._srch_13_rec(nd_13.lhs_13, val_13)
        return self._srch_13_rec(nd_13.rhs_13, val_13)
